### ISSUE TITLE
🐛 Bug Report: Cloudbuild Plugin limited to global scoped builds

### ISSUE THREAD

**Issue by vedantthapa** [backstage/backstage#21463]
### 📜 Description

As discussed in #19514 , cloudbuild only displays builds in the global scope. 

A potential fix might be use the https://cloud.google.com/build/docs/api/reference/rest/v1/projects.locations.builds/list endpoint to allow support for regional scoped builds as well.

cc: @PurseChicken 

### 👍 Expected behavior

Cloudbuild plugin should list builds in a regional scope.

### 👎 Actual Behavior with Screenshots

Request is completed, however, the response is empty due to the location of builds.
<img width="2741" alt="Screenshot 2023-11-21 at 10 06 22 PM" src="https://github.com/backstage/backstage/assets/43611693/5a41f6ad-7b3b-405f-a163-4fc7e577eda3">

### 👟 Reproduction steps

Follow the [plugin documentation](https://github.com/backstage/backstage/tree/master/plugins/cloudbuild).

### 🖥️ Your Environment

```sh
OS:   Darwin 23.1.0 - darwin/arm64
node: v20.5.0
yarn: 1.22.17
cli:  0.23.1 (installed)
backstage:  1.19.0

Dependencies:
  @backstage/app-defaults                                          1.4.4
  @backstage/backend-app-api                                       0.5.7, 0.5.8
  @backstage/backend-common                                        0.19.8, 0.19.9
  @backstage/backend-dev-utils                                     0.1.2
  @backstage/backend-openapi-utils                                 0.0.5
  @backstage/backend-plugin-api                                    0.6.6, 0.6.7
  @backstage/backend-tasks                                         0.5.11, 0.5.12
  @backstage/catalog-client                                        1.4.5, 1.4.6
  @backstage/catalog-model                                         1.4.3
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.1.5, 0.2.0
  @backstage/cli                                                   0.23.1
  @backstage/config-loader                                         1.5.2, 1.5.3
  @backstage/config                                                1.1.1
  @backstage/core-app-api                                          1.11.0
  @backstage/core-components                                       0.13.8, 0.13.7
  @backstage/core-plugin-api                                       1.8.0, 1.7.0
  @backstage/e2e-test-utils                                        0.1.0
  @backstage/errors                                                1.2.3
  @backstage/eslint-plugin                                         0.1.3
  @backstage/frontend-app-api                                      0.2.0
  @backstage/frontend-plugin-api                                   0.2.0, 0.3.0
  @backstage/integration-aws-node                                  0.1.7, 0.1.8
  @backstage/integration-react                                     1.1.20, 1.1.21
  @backstage/integration                                           1.7.1, 1.7.2
  @backstage/plugin-api-docs                                       0.9.12
  @backstage/plugin-app-backend                                    0.3.54
  @backstage/plugin-app-node                                       0.1.6
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.0
  @backstage/plugin-auth-backend-module-github-provider            0.1.3
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.3
  @backstage/plugin-auth-backend-module-google-provider            0.1.3
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.3
  @backstage/plugin-auth-backend                                   0.19.4
  @backstage/plugin-auth-node                                      0.4.0, 0.4.1
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.3
  @backstage/plugin-catalog-backend                                1.14.0
  @backstage/plugin-catalog-common                                 1.0.17, 1.0.18
  @backstage/plugin-catalog-graph                                  0.2.37
  @backstage/plugin-catalog-import                                 0.10.1
  @backstage/plugin-catalog-node                                   1.4.7, 1.5.0
  @backstage/plugin-catalog-react                                  1.9.1, 1.8.5
  @backstage/plugin-catalog                                        1.14.0
  @backstage/plugin-cloudbuild                                     0.3.27
  @backstage/plugin-events-node                                    0.2.15
  @backstage/plugin-github-actions                                 0.6.6
  @backstage/plugin-graphiql                                       0.2.55
  @backstage/plugin-kubernetes-backend                             0.13.1
  @backstage/plugin-kubernetes-common                              0.6.6, 0.7.1
  @backstage/plugin-kubernetes-node                                0.1.1
  @backstage/plugin-kubernetes-react                               0.1.1
  @backstage/plugin-kubernetes                                     0.10.3, 0.11.2
  @backstage/plugin-org                                            0.6.15
  @backstage/plugin-permission-common                              0.7.10, 0.7.9
  @backstage/plugin-permission-node                                0.7.17, 0.7.18
  @backstage/plugin-permission-react                               0.4.16, 0.4.17
  @backstage/plugin-proxy-backend                                  0.4.4
  @backstage/plugin-scaffolder-backend                             1.18.1
  @backstage/plugin-scaffolder-common                              1.4.2
  @backstage/plugin-scaffolder-node                                0.2.7
  @backstage/plugin-scaffolder-react                               1.5.6
  @backstage/plugin-scaffolder                                     1.15.1
  @backstage/plugin-search-backend-module-catalog                  0.1.10
  @backstage/plugin-search-backend-module-pg                       0.5.15
  @backstage/plugin-search-backend-module-techdocs                 0.1.10
  @backstage/plugin-search-backend-node                            1.2.10
  @backstage/plugin-search-backend                                 1.4.6
  @backstage/plugin-search-common                                  1.2.7, 1.2.8
  @backstage/plugin-search-react                                   1.7.1
  @backstage/plugin-search                                         1.4.1
  @backstage/plugin-tech-radar                                     0.6.9
  @backstage/plugin-techdocs-backend                               1.8.0
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.1
  @backstage/plugin-techdocs-node                                  1.9.0
  @backstage/plugin-techdocs-react                                 1.1.12
  @backstage/plugin-techdocs                                       1.8.0
  @backstage/plugin-user-settings                                  0.7.11
  @backstage/release-manifests                                     0.0.10
  @backstage/test-utils                                            1.4.4
  @backstage/theme                                                 0.4.4, 0.4.3
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.6, 1.0.7
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [x] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by PurseChicken** [comment]
@vedantthapa Thanks!

I'll work on a PR for this. We will likely need a new annotation to specify the region for the component.

**Comment by PurseChicken** [comment]
@vedantthapa 

Could you please let me know if the following method returns information about one of your builds outside of the global scope?

https://cloud.google.com/build/docs/api/reference/rest/v1/projects.builds/get?

Use your projectID and buildID for the parameters of "Try this Method".

Thanks!

**Comment by vedantthapa** [comment]
@PurseChicken

It returns an error:
```json
{
  "error": {
    "code": 404,
    "message": "Requested entity was not found.",
    "status": "NOT_FOUND"
  }
}
```

**Comment by PurseChicken** [comment]
> @PurseChicken
> 
> It returns an error:
> 
> ```json
> {
>   "error": {
>     "code": 404,
>     "message": "Requested entity was not found.",
>     "status": "NOT_FOUND"
>   }
> }
> ```

@vedantthapa 

Crud, I was afraid of that.

Can you use the following method instead?

https://cloud.google.com/build/docs/api/reference/rest/v1/projects.locations.builds/get?

In the name parameter use the following:

`projects/,your-project./locations/us-central1/builds/<your-build-id>`

**Comment by vedantthapa** [comment]
@PurseChicken, already did and it returns the relevant build :)

**Comment by PurseChicken** [comment]
@vedantthapa 

Thanks so much for confirming. In order to support builds outside the global scope we will need to modify all the Cloud Build queries in the plugin. Its going to take me a bit longer to get that done. It will likely need to wait until after my existing PR is approved for Cloud Build.

**Comment by camilaibs** [comment]
Hi @PurseChicken 👋🏻 , could you link this issue with the existing PR?

**Comment by PurseChicken** [comment]
@camilaibs Sure. Ill update the existing PR and link this issue. I'll do that early next week due to the US holiday.

Thanks!

**Comment by vinzscam** [comment]
Closed by #21397
